### PR TITLE
CXP-359 & CXP-367: Update error messages - Range

### PR DESCRIPTION
**Pull request by tomglvng**
EE --> https://github.com/akeneo/pim-enterprise-dev/pull/11126
Messages : 
 - The %attribute% attribute requires an equal or greater than %min_value% value.
 - The %attribute% attribute requires an equal or lesser than %max_value% value.

### CODE REVIEW

File: tests/back/Pim/Enrichment/Specification/Component/Product/Validator/Constraints/RangeValidatorSpec.php [hunk 1]
```diff
@@ -107,45 +107,46 @@ function it_does_not_validate_a_value_not_in_range(
...
-        $violation->setParameter('{{ min }}', 10)->shouldBeCalled()->willReturn($violation);
-        $violation->setParameter('{{ max }}', 100)->shouldBeCalled()->willReturn($violation);
-
-        $violation->setCode(Argument::any())->shouldBeCalled()->willReturn($violation);
+        $violation->atPath('data')->shouldBeCalled()->willReturn($violation);
+        $violation->setCode(Range::TOO_LOW_ERROR)->willReturn($violation);
         $violation->addViolation()->shouldBeCalled();
 
-        $this->validate(9.99999, $constraint);
+        $this->validate( 9.99999, $constraint);
```
Comment (romaindak) [inline tests/back/Pim/Enrichment/Specification/Component/Product/Validator/Constraints/RangeValidatorSpec.php:129]:
```suggestion
        $this->validate(9.99999, $constraint);
```

**Review by romaindak — APPROVED** [review]

**Review by RiffVentura — APPROVED** [review]

File: tests/back/Pim/Enrichment/Specification/Component/Product/Validator/Constraints/RangeSpec.php [hunk 2]
```diff
@@ -36,6 +36,12 @@ function it_has_an_invalid_number_message()
         $this->invalidMessage->shouldBe('The {{ attribute }} attribute requires a number, and the submitted {{ value }} value is not.');
     }
 
+    function it_has_an_min_message()
```
Comment (tomglvng) [inline tests/back/Pim/Enrichment/Specification/Component/Product/Validator/Constraints/RangeSpec.php:39]:
```suggestion
    function it_has_a_min_message()
```

File: tests/back/Pim/Enrichment/Specification/Component/Product/Validator/Constraints/RangeSpec.php [hunk 3]
```diff
@@ -36,6 +36,12 @@ function it_has_an_invalid_number_message()
...
+    {
+        $this->beConstructedWith(['max' => new \DateTime()]);
```
Comment (tomglvng) [inline tests/back/Pim/Enrichment/Specification/Component/Product/Validator/Constraints/RangeSpec.php:41]:
```suggestion
         $this->beConstructedWith(['min' => 1]);
```

File: src/Akeneo/Pim/Enrichment/Component/Product/Validator/Constraints/RangeValidator.php [hunk 4]
```diff
@@ -52,6 +53,10 @@ public function validate($value, Constraint $constraint)
                     ->addViolation();
                 break;
 
+            case is_numeric($value) && $value < $constraint->min:
```
Comment (pchasle) [inline src/Akeneo/Pim/Enrichment/Component/Product/Validator/Constraints/RangeValidator.php:56]: This doesn't work, `$constraint->min` can be null.

**Review by pchasle — CHANGES_REQUESTED** [review]

File: src/Akeneo/Pim/Enrichment/Component/Product/Validator/Constraints/RangeValidator.php (hunk 4 above)
Comment (pchasle) [reply src/Akeneo/Pim/Enrichment/Component/Product/Validator/Constraints/RangeValidator.php:56]: I don't think that you should use `validateData` either, this is defined for `Metric` or `ProductPrice` as specified in the comment (the `->atPath('data')` would not be valid for a simple scalar value).

I will suggest to do something like that https://github.com/akeneo/pim-community-dev/pull/14798/files#diff-6c4337f1523f1a32e45b45efe4b0a06a2b2c05a42f6ad44cb28ad65d72708440R26
To rebuild the constraint with our own message.

Comment (pchasle) [reply src/Akeneo/Pim/Enrichment/Component/Product/Validator/Constraints/RangeValidator.php:56]: You still need to keep your changes in the violation of `validateData` though for `Metric` and `ProductPrice` though :)

Comment (tomglvng) [reply src/Akeneo/Pim/Enrichment/Component/Product/Validator/Constraints/RangeValidator.php]: See changes in https://github.com/akeneo/pim-community-dev/pull/14811/commits/e584a25dea485a1f7f30dcef2dc6a5a474bfef65

File: src/Akeneo/Pim/Enrichment/Component/Product/Validator/Constraints/RangeValidator.php [hunk 5]
```diff
@@ -125,4 +102,51 @@ private function validateDateTime(\DateTimeInterface $dateTime, Constraint $cons
                 ->addViolation();
         }
     }
+
+    private function validateNumeric($value, Range $constraint): void
+    {
+        $violation = $this->buildRangeViolation($value, $constraint);
+
+        if ($violation instanceof ConstraintViolationBuilderInterface) {
+            $violation->addViolation();
```
[the remark this answers is not in the document]
Comment (tomglvng) [reply src/Akeneo/Pim/Enrichment/Component/Product/Validator/Constraints/RangeValidator.php:111]: Yep, atPath() is used when value comes from an object ; here, it is just a scalar.

**Review by romaindelouard — APPROVED** [review]

**Review by RiffVentura — APPROVED** [review]

**Review by pchasle — APPROVED** [review]

**Pull request merged**